lib/home_page.dart

```diff
@@ -8,15 +24,288 @@ class HomePage extends StatefulWidget {
+    ''').eq('read', false).neq('sender_id', profileId).order('created_at'),
+    );
+    items.addAll(Message.fromJsonList(messagesData));
+    final List<Map<String, dynamic>> rideEventsData = parseHelper.parseListOfMaps(
+      await SupabaseManager.supabaseClient.from('ride_events').select('''
+      *,
+      ride: ride_id(*,
+        rider: rider_id(*),
+        drive: drive_id(*,
+          driver: driver_id(*)
+          )
+      )
+      ''').eq('read', false).order('created_at'),
+    );
+    items.addAll(RideEvent.fromJsonList(rideEventsData).where((RideEvent rideEvent) => rideEvent.isForCurrentUser()));
+    items.sort((Model a, Model b) => b.createdAt!.compareTo(a.createdAt!));
+
+    setState(() {
+      _items = items;
+      _fullyLoaded = true;
+    });
+  }
+
   @override
   Widget build(BuildContext context) {
     return Scaffold(
       appBar: AppBar(
-        title: const Text('Home'),
+        title: Text(S.of(context).pageHomeTitle),
       ),
-      body: const Center(
-        child: Text('Home'),
+      body: Padding(
+        padding: const EdgeInsets.symmetric(
+          horizontal: 20,
+          vertical: 10,
+        ),
+        child: Column(
+          children: <Widget>[
+            const SizedBox(height: 30),
+            if (_fullyLoaded)
+              Expanded(
+                child: Container(
+                  width: MediaQuery.of(context).size.width,
+                  padding: const EdgeInsets.symmetric(vertical: 10, horizontal: 10),
+                  decoration: BoxDecoration(
+                    border: Border.all(color: Theme.of(context).primaryColor),
+                    borderRadius: const BorderRadius.all(Radius.circular(5.0)),
+                  ),
+                  child: _items.isNotEmpty
+                      ? ListView.separated(
+                          itemCount: _items.length,
+                          separatorBuilder: (BuildContext context, int index) {
+                            return const SizedBox(height: 12);
+                          },
+                          itemBuilder: (BuildContext context, int index) {
+                            return _buildWidget(_items[index], context);
+                          },
+                        )
+                      : Column(
+                          mainAxisAlignment: MainAxisAlignment.center,
+                          children: <Widget>[
+                            Image.asset(
+                              'assets/chat_shrug.png',
+                              scale: 8,
+                            ),
+                            const SizedBox(height: 16),
+                            Text(
+                              S.of(context).pageHomePageEmpty,
+                              style: Theme.of(context).textTheme.headline6,
+                            ),
+                            const SizedBox(height: 3),
+                          ],
+                        ),
+                ),
+              )
```

Comment:
Didn't we plan to put upcoming drives/rides here? Or is that going to be another PR?

Reply:
If so, I would argue that we should definitely put upcoming drives/rides separately from events, in which case the whole page will become scrollable because there's not enough space for another panel right now and we should definitely find another solution for the "scrollable panel" solution if we don't want scrollables inside of scrollables

Reply:
i don't think that to panels are a good option, we can include them and pin them at the top.